_posts/2025-04-19-navigating-pagination-with-hybrid-query.md

```diff
@@ -0,0 +1,229 @@
+
+### Results
+
+Our tests used an index containing 134 million documents (four copies of the `noaa_semantic_search` dataset).
+
+The following graph presents the results of running a hybrid query containing three BM25 subqueries: term, range, and date.
+
+![Performance-graph-of-hybrid-query-with-pagination](/assets/media/blog-images/2025-04-19-navigating-pagination-with-hybrid-query/Performance-graph-of-hybrid-query-with-pagination.png){:class="img-centered" width="500px"}
+
+The following graph presents the results of running a metric aggregation with a hybrid query containing BM25 three subqueries: term, range, and date.
+
+![metric-aggregation-performance-graph](/assets/media/blog-images/2025-04-19-navigating-pagination-with-hybrid-query/metric-aggregation-performance-graph.png){:class="img-centered" width="500px"}
+
+The following graph presents the CPU utilization results.
+
+![cpu-utilization](/assets/media/blog-images/2025-04-19-navigating-pagination-with-hybrid-query/cpu-utilization.png){:class="img-centered" width="500px"}
+
+### Performance summary
+
+The results show gradual, predictable performance changes:
+- No significant latency or CPU utilization spikes.
+- Latency metrics (p50, p90, p99, p100) and CPU utilization increase steadily with increasing pagination depth.
+
+## Best practices for using pagination_depth
+
+Our benchmarking shows no significant performance impact when fetching the second page, even with a `pagination_depth` of 10,000. However, both latency and CPU utilization increase with pagination depth. To optimize performance, consider the following recommendations.
+
+### Optimize pagination depth based on data distribution
+
+The efficiency of pagination depends on how evenly your data is distributed across shards. You can check this distribution using the `_cat/shards` API. If the documents are equally distributed amongst shards, then even a smaller `pagination_depth` can provide more hybridized search results per subquery.
+
+For example:
+* With `pagination_depth = 20` across three shards, you can hybridize up to 60 results per subquery.
+* This assumes each shard contains at least 20 results per subquery (thus, the data is evenly distributed).
+* If there are no duplicate entries between subqueries, you can paginate through up to 120 hybridized search results.
+
+### Handle uneven data distribution
+
+Custom routing can lead to uneven data distribution across shards, resulting in:
+* Some shards returning zero results, reducing the total number of hybridized results.
+
+### Some shards may contain all the relevant results
```

Comment:
This is not a best practice/recommendation like the preceding two.

Reply:
@kolchfa-aws @natebower actually this section should be a subsection of the earlier one Manage uneven data distribution. Some shards may contain all the relevant results section contains the explanation what is written in the earlier section with the example.

Reply:
OK then, I'll just remove the heading and make it a paragraph in that section.